File: ski.py
# Ivy Loi
# Homework 3, Problem 1a

def skiCost(days, skiRental, skiPurchase):
    rentTot = days * skiRental
    if rentTot < skiPurchase:
        return True
    else:
        return False



def main():
    dailySkiRental = 100
    purchaseSki = 250
    sampleDays = 5
    print("The cost to rent skis for", sampleDays, "days is $", sampleDays * dailySkiRental)
    if skiCost(sampleDays, dailySkiRental, purchaseSki) == True:
        print("It is cheaper to rent skis for", sampleDays, "days.")
    else:        
        print("It is cheaper to buy skis than to rent for $", purchaseSki)
    
    Days = int(input("How many days do you want to rent skis? "))
    if skiCost(Days, dailySkiRental, purchaseSki) == True:
        print("It is cheaper to rent skis for", Days, "days.")
    else:        
        print("It is cheaper to buy skis than to rent for $", purchaseSki)


if __name__ == "__main__":
    main()

File: ski.cpp
/* Name: Ivy Loi
Date: 4/24/2026
Homework 3, Problem 1a
CPSC 335 section 10
*/
#include <iostream>

bool skiCost(int days, int skiRental, int skiPurchase) {
    // Calculate the total cost of renting skis for the given number of days
    int totalRentalCost = 0;
    for (int i = 0; i < days; i++) {
        totalRentalCost += skiRental;
    }
    // Compare the total rental cost with the purchase cost
    if (totalRentalCost < skiPurchase) {
        // If renting is cheaper, return true
        return true;
    } else {
        return false;
    }
}

int main() {
    int days = 5;
    int skiRental = 100;
    int skiPurchase = 250;
    int rentCost = days * skiRental;

    std::cout << "Cost of renting skis is $" << skiRental << std::endl;
    std::cout << "Cost of purchasing skis: $" << skiPurchase << std::endl;
    std::cout << "You are planning to go skiing for " << days << " days." << std::endl;
    if (skiCost(days, skiRental, skiPurchase)) {
        std::cout << "It is cheaper to rent skis for " << days << " days"
                  << " for $" << rentCost << std::endl;
    } else {
        std::cout << "It is cheaper to purchase skis for $" << skiPurchase << "." << std::endl;
    }

    return 0;
}